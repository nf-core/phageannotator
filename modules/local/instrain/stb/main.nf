process INSTRAIN_STB {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'quay.io/biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.stb")  , emit: stb
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    create_instrain_stb.py \\
        -f $fasta \\
        -o ${prefix}.stb

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.stb

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        pandas: \$(echo \$(pandas_version.py 2>&1))
    END_VERSIONS
    """
}
