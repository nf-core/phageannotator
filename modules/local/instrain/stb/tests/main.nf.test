nextflow_process {

    name "Test Process: INSTRAIN_STB"
    script "../main.nf"
    process "INSTRAIN_STB"


    test("genome.fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'reference_fasta' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("genome.fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'reference_fasta' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
