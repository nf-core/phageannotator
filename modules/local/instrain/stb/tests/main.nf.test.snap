{
    "genome.fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,8f9e23730f7fda6be33ce21216fb7f2d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ],
                "stb": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,8f9e23730f7fda6be33ce21216fb7f2d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:10:31.746010063"
    },
    "genome.fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ],
                "stb": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:10:42.221200638"
    }
}
